coalib/misc/Shell.py

```diff
@@ -99,11 +99,11 @@ def get_shell_type():  # pragma: no cover
              Powershell is detected, "cmd" if command prompt is been
              detected or "sh" if it's neither of these.
     """
-    out_hostname, _ = run_shell_command(["echo", "$host.name"], shell=True)
-    if out_hostname.strip() == "ConsoleHost":
+    out = run_shell_command("echo $host.name", shell=True)[0]
+    if out.strip() == "ConsoleHost":
         return "powershell"
-    out_0, _ = run_shell_command(["echo", "$0"], shell=True)
-    if out_0.strip() == "" and out_0.strip() == "":
+    out = run_shell_command("echo $0", shell=True)[0]
+    if out.strip() == "$0":
```

Comment:
you're doing a lot of small fixes on this function which is IMO why it's so hard to find a commit message that fits, I think this needs a splitup